code/modules/submaps/submap_archetype.dm

```diff
@@ -11,14 +11,24 @@
 	/// Whether the job preferences for this submap archetype are collapsed by default.
 	var/default_to_hidden = TRUE
 
+var/global/list/submaps_by_name
 /decl/submap_archetype/validate()
 	. = ..()
-	if(!descriptor)
-		. += "no descriptor set"
+	if(!name)
+		. += "no name set"
+	else
+		if(!global.submaps_by_name)
+			global.submaps_by_name = list()
+			for(var/decl/submap_archetype/submap as anything in decls_repository.get_decls_of_type_unassociated(/decl/submap_archetype))
```

Comment:
i wonder if this is actually necessary? we're already looping over all submap archetypes in the validation test, so theoretically it could just be
```dm
		if(global.submaps_by_name[name])
			. += "name '[name]' conflicts with submap type '[global.submaps_by_name[name]]'"
		else
			global.submaps_by_name[name] = type
```
and obviously submaps_by_name would have to always be a list for this code to work exactly as i wrote it
that said i think it should be a static var to avoid someone trying to access it outside of validation